### ISSUE TITLE
Service Dependencies (gdrive etc) should be implemented against interfaces to ease testing

### ISSUE THREAD

**Issue by joefroh** [KyleBrownCS/-r-winnipegjets-scripts#19]
Right now the bot uses a couple of wrapper classes that ease data i/o from services (secrets, gdrive and so on). That's good, but mocking them out for testing will get a bit harry if we don't explicitly define the interface. I am going to hand roll it for now, but interfaces would make this tidier moving forward and centralize where the interface for the service is defined so both the mock and the actual class have one source of truth for the api contract

**Comment by joefroh** [comment]
I'm gonna claim this one unless you feel like you wanna do it. Just came up as I started writing out the tests, and I don't wanna hold up my PR on it.

**Comment by KyleBrownCS** [comment]
Go for it! Was definitely on my road map, the drive_manager.py isn't very "specialized" and handles way too much stuff outside it's original scope

**Comment by joefroh** [comment]
👍 let me know if I’m over doing it! I’m just having fun contributing.

________________________________

**Comment by KyleBrownCS** [comment]
Not at all! Very happy to have a second dev on the project. Hoping to learn as much as I can from you and improve this project so I don't need to do as much manual configuring/patching as there is currently!

I have to apologize that I didn't plan this project out more, wouldn't have so many headaches. Kind of just dove in and wrote it. Many regrets

**Comment by joefroh** [comment]
Its all good man :) The best projects are organic, and its a testament to the utility of the project that something you hacked away on has grown this much in scope :) Trust me, this is not bad at all, I have seen A LOT WORSE.

All things considered this is really tidy and well commented, I can get in, refactor and get out without much trouble, and you being supportive of the changes and helping me understand stuff has been awesome!
